### ISSUE TITLE
pxoxying over gnunet or i2p, or bridging between i2p torrents and clear torrents

### ISSUE THREAD

**Issue by je-vv** [rakshasa/rtorrent#1168]
I'm wondering if eventually rtorrent would be able to use gnunet or i2p as some sort or proxy mechanisms, so that still torrents from clear/regular trackers, but through gnunet or i2p.  Or if not possible at all, then become able to bridge torrent between i2p torrents trackers and clear trackers, being able to hooke to i2p for the 1st ones.  I understand Vuze does the latest, and some other client, but to me, there's nothing like rtorrent, not requiring a web interface neither a gui to monitor it, just a terminal or a terminal emulator, or SSH if rtorrent is running remotely....

Sorry if this goes way beyond rtorrent's scope.  BTW, I do use:

```
protocol.encryption.set = allow_incoming,try_outgoing,enable_retry
```

And I guess that might help, but it might be far from enough now a days.  I'd like to use:

```
protocol.encryption.set = require,require_RC4,allow_incoming,try_outgoing
```

But then nothing gets downloaded if used....  If nothing else is possible, is there any recommendation for `protocol.encryption` to get better protected, without a VPN (VPNs are being asked to block traffic, and to provide information about users now a days)?

Thanks !

**Comment by kannibalox** [comment]
The torrent protocol's encryption provides very little actual protection, and is mostly intended to bypass ISP throttling. "Obfuscation" would be a better word for it, and that applies to all clients, not just rTorrent. Your first settings were fine.

**Comment by je-vv** [comment]
@kannibalox you're right, and that's the motivation for looking true alternatives to VPN, which doesn't anonymize either, and providers are getting forced to log and provide logs, and so on...  I have no other expectations than obfuscation for the torrenting encryption.  I've read about i2p trackers, but how limited their contents are, and also few current clients.  I'd prefer something that might be used more like to tor for torrenting, but there aren't alternatives, except perhaps i2p, gnunet and similar.  Not sure if there are efforts in that direction.

That said, being that there are clients actually bridging between both i2p trackers and clear trackers, or if you prefer i2p trackers clients, not sure if rtorrent devs have explore such possibility, or would get any interest on that.  Unfortunately such current clients are nothing like rtorrent (and honestly, I know nothing as light and low demanding to monitor remote setups, as mentioned, I just need a ssh connection, and I can handle everything remotely).  So that's why I opened this issue.
